evaluator.measure-hapi/src/main/java/org/opencds/cqf/cql/evaluator/measure/r4/R4MeasureReportAggregator.java

```diff
@@ -68,22 +71,46 @@ protected void mergeContained(MeasureReport carry, MeasureReport current) {
             return;
         }
 
-        List<String> resourceIds = new ArrayList<>();
+        Map<String, Resource> resourceMap = new HashMap<>();
 
         for (Resource resource : carry.getContained()) {
             if (resource.hasId()) {
-                resourceIds.add(resource.getId());
+                resourceMap.put(resource.getId(), resource);
             }
         }
+        carry.getContained().clear();
 
         for (Resource resource : current.getContained()) {
-            if (resource.hasId()) {
-                if (!resourceIds.contains(resource.getId())) {
-                    resourceIds.add(resource.getId());
-                    carry.getContained().add(resource);
+            if (resource.hasId() && resource.getResourceType().equals(ResourceType.List)) {
+                if (resourceMap.containsKey(resource.getId())) {
```

Comment:
The lists here may have different Ids. You'll have to pick the correct list based on the population list reference in the population group.